src/iidm/DanglingLine.cpp

```diff
@@ -15,15 +15,21 @@ namespace powsybl {
 namespace iidm {
 
 DanglingLine::DanglingLine(VariantManagerHolder& network, const std::string& id, const std::string& name, bool fictitious,
-                           double p0, double q0, double r, double x, double g, double b, const std::string& ucteXnodeCode) :
+                           double p0, double q0, double r, double x, double g, double b, const std::string& ucteXnodeCode,
+                           std::unique_ptr<Generation>&& generation) :
```

Comment:
Probably that storing the generation in an optional rather than a unique_ptr should be simpler.

Reply:
I would try to update the implementation with an optional. I would also try it for the ShuntCompensatorModel.

If it works well, I will create an issue to change all our unique_ptr usage by optional